src/main/java/carpet/mixins/StructureBlockScreen_structureBlockLimitMixin.java

```diff
@@ -0,0 +1,29 @@
+package carpet.mixins;
+
+import carpet.CarpetSettings;
+import carpet.network.ClientNetworkHandler;
+import net.minecraft.block.entity.StructureBlockBlockEntity;
+import net.minecraft.client.gui.screen.ingame.StructureBlockScreen;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(StructureBlockScreen.class)
+public abstract class StructureBlockScreen_structureBlockLimitMixin
+{
+    @Shadow @Final private StructureBlockBlockEntity structureBlock;
+
+    @Inject(method = "method_2516", at = @At(value = "TAIL"), locals = LocalCapture.CAPTURE_FAILHARD)
+    private void sendCarpetStructureBlockPacket(StructureBlockBlockEntity.Action action, CallbackInfoReturnable<Boolean> cir, BlockPos blockPos, BlockPos blockPos2)
+    {
+        if (CarpetSettings.structureBlockLimit != CarpetSettings.VANILLA_STRUCTURE_BLOCK_LIMIT)
```

Comment:
we really need to send this is the limit is > 127

Reply:
In current `UpdateStructureBlockC2SPacketMixin`, there isn't any implementation that stops vanilla `net.minecraft.network.packet.c2s.play.UpdateStructureBlockC2SPacket#read` method clamping the byte value in the packet. 

```java
// in net.minecraft.network.packet.c2s.play.UpdateStructureBlockC2SPacket#read
this.offset = new BlockPos(MathHelper.clamp(buf.readByte(), -48, 48), MathHelper.clamp(buf.readByte(), -48, 48), MathHelper.clamp(buf.readByte(), -48, 48));
this.size = new BlockPos(MathHelper.clamp(buf.readByte(), 0, 48), MathHelper.clamp(buf.readByte(), 0, 48), MathHelper.clamp(buf.readByte(), 0, 48));
```

So If this if judgement is changed to testing limit > 127, then some extra mixins are needed to prevent vanilla server to clamp our offset/size value into vanilla 48 range

Reply:
Some extra mixins can be some `@ModifyConstant` to change those `-48`, `48` into `CarpetSettings.structureBlockLimit`. An example can be found [here](https://github.com/TISUnion/Carpet-TIS-Addition/blob/1.16.5/src/main/java/carpettisaddition/mixins/rule/structureBlockLimit/UpdateStructureBlockC2SPacketMixin.java#L13-L32)

Reply:
So should I change the if judgement to > 127 and implement these extra mixins?

Reply:
true - I will need to change that logic I guess